kaolin/ops/conversions/pointcloud.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2019-2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+__all__ = ['pointclouds_to_voxelgrids']
+
+def _base_points_to_voxelgrids(points, resolution, return_sparse=False):
+    r"""Converts points to voxelgrids. This is the base function for both trianglemeshes_to_voxelgrids
+    and pointclouds_to_voxelgrids. For point cloud, the points are the original points. For Triangle Mesh,
+    the points are sampled along the edges. The points are normalized into range [0, 1].
```

Comment:
I'm not sure we should reference mesh behavior, the behavior that you described here ("For Triangle Meshes, the points are samples along the edges") has nothing to do with this function, isn't it? If you changed the way you sampled points along the edges this function would still be the same the description would be false, which is difficult to maintain.

Also, are the points normalized? It seemed to be that only points in the range [0, 1] (before doing anything to them) were sampled.

Reply:
Yes, the points are normalized. So will it be sufficient to say that the function converts normalized points in the range [0, 1] to voxelgrid?

Reply:
My understanding is that the point cloud MAY not be normalized, but regardless of that this base function will ONLY voxelize points that are in the range [0, 1]. Am I correct?